### PR TITLE
add vbk context validation

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: src/mempool.cpp [hunk 1]
```diff
@@ -446,4 +451,18 @@ bool MemPool::checkContextually<ATV>(const ATV& atv, ValidationState& state) {
   return true;
 }
 
+template <>
+bool MemPool::checkContextually<VbkBlock>(const VbkBlock& blk,
+                                          ValidationState& state) {
+  if (tree_->vbk().getBlockIndex(blk.previousBlock) == nullptr &&
+      get<VbkBlock>(blk.previousBlock) == nullptr) {
+    return state.Invalid(
+        "potential-stale-vbk-block",
```
Comment (Warchant) [inline src/mempool.cpp:460]:
```suggestion
        "bad-prev",
```

File: src/mempool.cpp [hunk 2]
```diff
@@ -446,4 +451,18 @@ bool MemPool::checkContextually<ATV>(const ATV& atv, ValidationState& state) {
...
+        fmt::sprintf("Block=%s does not connect to the known chains",
```
Comment (Warchant) [inline src/mempool.cpp:461]:
```suggestion
        fmt::sprintf("Block=%s does not connect to known VBK tree",
```

**Review by Warchant — APPROVED** [review]
Make these fixes and then can be merged

File: src/mempool.cpp (hunk 1 above)
Comment (Mr-Leshiy) [reply src/mempool.cpp]: fixed

File: src/mempool.cpp (hunk 2 above)
Comment (Mr-Leshiy) [reply src/mempool.cpp]: fixed

**Pull request merged**